### PR TITLE
docs/psoc6: Adding hardware documentation.

### PR DESCRIPTION
**Pull request by NikhitaR-IFX**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description

Extended documentation support with:
1. Added pinout diagrams for both boards.
2. Sections per board with resource links
3. Board overview and distinguishing feature

Find the documentation link for review [here](https://ifx-micropython.readthedocs.io/en/docs-hw-section/psoc6/general.html)

*Note: We still have to add the final versions of the pinout after:
1. Right versioning is added
2. BLE pinout is verified

Please use this PR to review the content and overall alignment of sections etc.

### CODE REVIEW

**Review by jaenrig-ifx — APPROVED** [review]

**Comment by ederjc** [comment]
Can we put the board overview like this?
| Board Name | GPIOs | Connectivity | Special Features |
| ------------- | ------------- | ------------- | ------------- |
| CY8CPROTO-062-4343W  | 58 | WiFi, BT  | microSD card slot, external flash memory, thermistor, user button |
| CY8CPROTO-063-BLE  | 32 | BLE | user button, compact form-factor

I would also not put the features in the table which we currently not support in our port.

I think the pinout diagrams are still too small. Maybe I can find a solution to make them more narrow.

**Comment by NikhitaR-IFX** [comment]
> Can we put the board overview like this?
> 
> Board Name	GPIOs	Connectivity	Special Features
> CY8CPROTO-062-4343W	58	WiFi, BT	microSD card slot, external flash memory, thermistor, user button
> CY8CPROTO-063-BLE	32	BLE	user button, compact form-factor
> I would also not put the features in the table which we currently not support in our port.
> 
> I think the pinout diagrams are still too small. Maybe I can find a solution to make them more narrow.

Thanks for the review. I can sure make these changes. I agree for connectivity column. I feel why GPIO's are special here? Then why not add serial communication interfaces?

**Review by IFX-Anusha — APPROVED** [review]

File: extmod/machine_i2c.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
```
Comment (jaenrig-ifx) [inline extmod/machine_i2c.h:2]: This file has to be removed.

**Review by jaenrig-ifx — CHANGES_REQUESTED** [review]
Accidentally added file to extmod.

File: extmod/machine_i2c.h (hunk 1 above)
Comment (NikhitaR-IFX) [reply extmod/machine_i2c.h:2]: Do we need to add anything more while removing this? Just removing this file makes the build fail.

Comment (NikhitaR-IFX) [reply extmod/machine_i2c.h]: Done

**Review by jaenrig-ifx — APPROVED** [review]

**Review by IFX-Anusha — APPROVED** [review]

**Review by ederjc — APPROVED** [review]
👍

**Pull request merged**